method.rb

```diff
@@ -0,0 +1,190 @@
+module Enumerable
+  def my_each
+    return 'no block given' unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return 'no block given' unless block_given?
+
+    i = 0
+    while i < size
+      yield self[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return 'no block given' unless block_given?
+
+    output_array = []
+    i = 0
+    while i < size
+      output_array << self[i] if (yield i) == true
+      i += 1
+    end
+    output_array
+  end
+
+  def my_all?
+    return 'no block given' unless block_given?
+
+    status_cumulator = true
+    i = 0
+    loop do
+      status_cumulator &&= yield i
+      i += 1
+      break if i == size || status_cumulator == false
+    end
+    status_cumulator
+  end
+
+  def my_any?
+    return 'no block given' unless block_given?
+
+    status_cumulator = false
+    i = 0
+    loop do
+      status_cumulator ||= yield i
+      i += 1
+      break if i == size || status_cumulator == true
+    end
+    status_cumulator
+  end
+
+  def my_none?
+    return 'no block given' unless block_given?
+
+    status_cumulator = true
+    i = 0
+    loop do
+      status_cumulator &&= yield i
+      i += 1
+      break if i == size || status_cumulator == false
+    end
+    !status_cumulator
+  end
+
+  def my_count
+    return size unless block_given?
```

Comment:
* [ ] There should be a working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- ```p [1, 2, 4, 2].my_count(2)``` should return  same result as ```p [1, 2, 4, 2].count(2)```